packages/grafana-e2e/src/flows/deleteDataSource.ts

```diff
@@ -1,13 +1,21 @@
 import { e2e } from '../index';
 import { fromBaseUrl } from '../support/url';
 
-export const deleteDataSource = (dataSourceName: string) => {
-  e2e().logToConsole('Deleting data source with name:', dataSourceName);
-  e2e().request('DELETE', fromBaseUrl(`/api/datasources/name/${dataSourceName}`));
+export interface DeleteDataSourceConfig {
+  id: string;
+  name: string;
+}
+
+export const deleteDataSource = ({ id, name }: DeleteDataSourceConfig) => {
+  e2e().logToConsole('Deleting data source with name:', name);
+
+  // Avoid datasources page errors
+  e2e.pages.Home.visit();
+  e2e().request('DELETE', fromBaseUrl(`/api/datasources/name/${name}`));
 
   /* https://github.com/cypress-io/cypress/issues/2831
```

Comment:
I think we can delete this commented out section now.

Reply:
Same question here as from the other one.

Reply:
My thought was that we can have a specific test that tests removal from the UI and let this be just a request. WDYT?

Reply:
That does make good sense. Although, we're already doing through the UI to add a data source, so it might be less complicated to use the same function everywhere.

Reply:
There is also the time aspect, deleting using a request will be xyz times faster than deleting from the UI and we want this to be as fast as possible.

Reply:
Right. Okay, I think the function should have both to fit the needs of the situation. I've added a `quick` option to the config.

Reply:
Awesome!